src/charm.py

```diff
@@ -51,15 +54,18 @@ def __init__(self, *args):
             self.on["ingress-auth"].relation_changed,
             self.on["oidc-client"].relation_changed,
             self.on["client-secret"].relation_changed,
+            self.on[OIDC_PROVIDER_INFO_RELATION].relation_changed,
+            self.on[OIDC_PROVIDER_INFO_RELATION].relation_broken,
         ]:
             self.framework.observe(event, self.main)
+        self.framework.observe(self._dex_oidc_config_requirer.on.updated, self.main)
 
         self._logging = LogForwarder(charm=self)
 
     def main(self, event):
         try:
-            self._check_public_url()
             self._check_leader()
+            self._check_dex_oidc_config_relation()
```

Comment:
Since this returns data, we could save them and use them later instead of calling `get_data()` twice.

Reply:
`self._check_dex_oidc_config_relation()` is not supposed to return anything, it will just make a quick check and ensure the relation is present and it has data. The reason why `get_data` is called there is because the requirer object currently does not provide any other way to make those checks. I guess it is a limitation of the library, perhaps we could make [this](https://github.com/canonical/dex-auth-operator/blob/main/lib/charms/dex_auth/v0/dex_oidc_config.py#L238) method public so the requirer code can call it.

That being said, we have two options:

1. Leave it as is and just use it as a check
2. Change the library to make this method public and refactor this PR.

wdyt?

Reply:
After a discussion with @orfeas-k we think this is a limitation of the library and it will be worked on https://github.com/canonical/dex-auth-operator/issues/215. After that change, we can refactor this method to remove the awkward call to `get_data()`. For now it has been decided to leave it as is until the library is changed.